tomcat-11: add tomcat 11.0.2

iigs opened the pull request:
Uses JDK 21.

iigs committed:
tomcat-11: add tomcat 11.0.2

mtelka commented on components/web/tomcat-11/Makefile:
```diff
@@ -0,0 +1,85 @@
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/ant.mk
+include $(WS_MAKE_RULES)/ips.mk
```
Please convert this to use `common.mk`.

iigs replied:
ant does not run if these lines are changed to common.  Please provide a followup PR if you can figure out how to use common.mk

mtelka replied:
Did you try with the `BUILD_STYLE` set to `ant`?

mtelka commented on components/web/tomcat-11/Makefile:
```diff
@@ -0,0 +1,85 @@
+COMPONENT_ARCHIVE_HASH=	\
+	sha256:94885a2591a1427dabae36cf2dc8d1839d25b05a3aee32aab9721431f6c1032c
+COMPONENT_ARCHIVE_URL=	https://archive.apache.org/dist/tomcat/tomcat-11/v$(COMPONENT_VERSION)/src/$(COMPONENT_ARCHIVE)
+COMPONENT_BUGDB=	utility/tomcat
```
Please remove this line.  It is not needed.

mtelka commented on components/web/tomcat-11/Makefile:
```diff
@@ -0,0 +1,85 @@
+
+CLASSPATH=$(PROTO_DIR)/lib:$(PROTO_DIR)/bin
+
+ifeq ($(strip $(USERLAND_ARCHIVES)),)
```
Why is this needed?

iigs replied:
I don't know.  It  was in the existing tomcat Makefile and I just ignored it.  I removed this section.  Please test to see if anything breaks.

mtelka commented on components/web/tomcat-11/Makefile:
```diff
@@ -0,0 +1,85 @@
+
+# Tomcat tests doesn't like cloning.
+CLONEY=echo
+COMPONENT_PRE_BUILD_ACTION= \
```
Please use `CLONEY_MODE` and set it to `copy` instead of this.

iigs replied:
removing CLONEY=echo and adding CLONEY_MODE=copy doesn't work.

mtelka replied:
What about something like this?
```
CLONEY_ARGS += CLONEY_MODE=copy
```

mtelka replied:
FYI, with #20305 merged the plain `CLONEY_MODE = copy` should start to work too.

iigs commented:
Updated Makefile.  gmake build still succeeds.

mtelka commented on components/web/tomcat-11/Makefile:
```diff
@@ -0,0 +1,79 @@
+CLASSPATH=$(PROTO_DIR)/lib:$(PROTO_DIR)/bin
+
+# Tomcat tests doesn't like cloning.
+# Marcel thinks this is the right approach
```
WTF is this?

mtelka commented on components/web/tomcat-11/Makefile:
```diff
@@ -0,0 +1,79 @@
+COMPONENT_NAME=		tomcat
+COMPONENT_VERSION=	11.0.2
+COMPONENT_PROJECT_URL=	https://tomcat.apache.org/
+COMPONENT_SRC_NAME=	apache-tomcat
```
You can set `COMPONENT_NAME` to `apache-tomcat` and use `COMPONENT_NAME` instead of `COMPONENT_SRC_NAME` (to save one line in the `Makefile` :-)).

mtelka commented on components/web/tomcat-11/Makefile:
```diff
@@ -0,0 +1,79 @@
+test:   $(NO_TESTS)
+
+# Build dependencies
+REQUIRED_PACKAGES += developer/build/ant
```
FYI, with #20305 merged this line will no longer be needed.

iigs commented:
* Now uses common.mk. Removed build: , install: , and test: sections.
* Changed line CLONEY=echo to: CLONEY_ARGS += CLONEY_MODE=copy
* Removed unneeded comments.  Added relevant comments.
gmake build still works.
